File: src/Utilities/Buildtools/Configuration/UIXConfiguration.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Buildtools.Configuration
{
    public class UIXConfiguration
    {
        public UIXConfiguration(Buildtools buildtools)
        {
            fileSystem.path.jsRootPath = buildtools.ServerFileStructure.JavaScriptRootPath;
        }

        public string buildTime = DateTime.UtcNow.ToString("o");
        public FileSystem fileSystem = new FileSystem();

        public class FileSystem
        {
            public Path path = new Path();
            public FileName fileName = new FileName();

            public class Path
            {
                public string jsRootPath = "/js/";
            }

            public class FileName
            {
                public string uix = "UIX.js";
                public string ajaxWorker = "UIX.AjaxWorker.js";
                public string serviceWorker = "UIX.ServiceWorker.js";
            }
        }
    }
}

File: src/Utilities/BuildTools/Configuration/UIXConfiguration.ts
/// <reference path="BuildToolsConfiguration.ts" />

namespace BuildTools.Configuration{

    export class UIXConfiguration{

        public constructor(buildToolsConfiguration:BuildToolsConfiguration){
            this.fileSystem.path.jsRootPath = buildToolsConfiguration.serverFileStructure.javaScriptRootPath;
        }

        public buildTime = new Date().toISOString();
        
        public fileSystem = {
            path: {
                jsRootPath: "/js/"
            },
            fileName: {
                uix: "UIX.js",
                ajaxWorker: "UIX.AjaxWorker.js",
                serviceWorker: "UIX.ServiceWorker.js"
            }
        };
    }
}